test/steps/servicepage.js

```diff
@@ -0,0 +1,41 @@
+/* @flow */
+
+"use strict";
+
+import assert from "assert";
+import Yadda from 'yadda';
+import _ from 'underscore';
+import { By } from 'selenium-webdriver';
+
+import dictionary from "../support/dictionary";
+import unpromisify from "../support/yadda-promise";
+
+async function asyncFilter(arr, check) {
+    var results = arr.map(check);
+    results = await Promise.all(results);
+    return _.zip(results, arr)
+        .filter((e) => e[0])
+        .map((e) => e[1]);
```

Comment:
You can do this in a single map and a comprehension (comprehension syntax may be wrong).

```
var results = arr.map(element => [element, check(element)]);
return [for (element, check) of results if (check) element];
```